src/quo/components/share/share_qr_code/view.cljs

```diff
@@ -1,46 +1,243 @@
+        (cycle [[line] [space]])))
+
+(defn- header
+  [{:keys [share-qr-type on-info-press on-legacy-press on-multichain-press]}]
+  [rn/view {:style style/header-container}
+   [tab/view
+    {:accessibility-label         :share-qr-code-legacy-tab
+     :id                          :wallet-legacy-tab
+     :active-item-container-style style/header-tab-active
+     :item-container-style        style/header-tab-inactive
+     :size                        24
+     :active                      (= :wallet-legacy share-qr-type)
+     :on-press                    on-legacy-press}
+    "Legacy"]
+   [rn/view {:style style/space-between-tabs}]
+   [tab/view
+    {:accessibility-label         :share-qr-code-multichain-tab
+     :id                          :wallet-multichain-tab
+     :active-item-container-style style/header-tab-active
+     :item-container-style        style/header-tab-inactive
+     :size                        24
+     :active                      (= :wallet-multichain share-qr-type)
+     :on-press                    on-multichain-press}
+    "Multichain"]
+   [rn/pressable
+    {:accessibility-label :share-qr-code-info-icon
+     :style               style/info-icon
+     :on-press            on-info-press
+     :hit-slop            6}
+    [icon/icon :i/info
+     {:size  20
+      :color style/info-icon-color}]]])
+
+(defn- info-label
+  [share-qr-code-type]
+  [text/text {:size :paragraph-2 :weight :medium :style style/title}
+   (if (= share-qr-code-type :profile)
+     (i18n/label :t/link-to-profile)
+     (i18n/label :t/wallet-address))])
+
+(defn- info-text
+  [{:keys [width on-press on-long-press ellipsize?]} qr-data-text]
+  [rn/pressable
+   {:accessibility-label :share-qr-code-info-text
+    :style               (style/data-text width)
+    :on-press            on-press
+    :on-long-press       on-long-press}
+   [text/text
+    (cond-> {:size   :paragraph-1
+             :weight :monospace}
+      ellipsize? (assoc :number-of-lines 1
+                        :ellipsize-mode  :middle))
+    qr-data-text]])
+
+(defn- share-button
+  [{:keys [alignment on-press]}]
+  [rn/view {:style (style/share-button-container alignment)}
+   [button/button
+    {:icon-only?          true
+     :type                :grey
+     :background          :blur
+     :size                style/share-button-size
+     :accessibility-label :link-to-profile
+     :on-press            on-press}
+    :i/share]])
+
+(defn- network-colored-text
+  [network-short-name]
+  [text/text {:style (style/network-short-name-text network-short-name)}
+   (str network-short-name ":")])
+
+(defn- wallet-multichain-colored-address
+  [full-address]
+  (let [[networks address]  (as-> full-address $
+                              (string/split $ ":")
+                              [(butlast $) (last $)])
+        ->network-hiccup-xf (map #(vector network-colored-text %))]
+    (as-> networks $
+      (into [:<>] ->network-hiccup-xf $)
+      (conj $ address))))
```

Comment:
@OmarBasem  you recently refactored this behaviour, here I just added another implementation of it. It doesn't need to set `^:key` for the text nodes, you could take it in consideration while abstracting the general solution.

Reply:
@ulisesmac can you please show me how to make that work for the implementation in https://github.com/status-im/status-mobile/pull/17732/files/cd03a2349f9335a5aca623cb8212b45d3c4b5b6d#diff-e3dc7e8c9c9d0e79e3e3ef677ba5e72ebbf9fe1b8ceee3adf30bd08ff17fee62 In my implementation the address is a component with props. I think the one here is just text?